diff --git a/node_modules/monaco-editor/esm/vs/editor/contrib/inlayHints/browser/inlayHintsController.js b/node_modules/monaco-editor/esm/vs/editor/contrib/inlayHints/browser/inlayHintsController.js
index ebc9d1a..4066946 100644
--- a/node_modules/monaco-editor/esm/vs/editor/contrib/inlayHints/browser/inlayHintsController.js
+++ b/node_modules/monaco-editor/esm/vs/editor/contrib/inlayHints/browser/inlayHintsController.js
@@ -212,7 +212,7 @@ let InlayHintsController = class InlayHintsController {
         }, this._debounceInfo.get(model));
         this._sessionDisposables.add(scheduler);
         this._sessionDisposables.add(toDisposable(() => cts?.dispose(true)));
-        scheduler.schedule(0);
+        queueMicrotask(() => scheduler.onTimeout());
         this._sessionDisposables.add(this._editor.onDidScrollChange((e) => {
             // update when scroll position changes
             // uses scrollTopChanged has weak heuristic to differenatiate between scrolling due to
@@ -223,9 +223,8 @@ let InlayHintsController = class InlayHintsController {
         }));
         this._sessionDisposables.add(this._editor.onDidChangeModelContent((e) => {
             cts?.cancel();
-            // update less aggressive when typing
-            const delay = Math.max(scheduler.delay, 1250);
-            scheduler.schedule(delay);
+            scheduler.cancel();
+            queueMicrotask(() => scheduler.onTimeout());
         }));
         // mouse gestures
         this._sessionDisposables.add(this._installDblClickGesture(() => scheduler.schedule(0)));
diff --git a/node_modules/monaco-editor/esm/vs/editor/contrib/semanticTokens/browser/documentSemanticTokens.js b/node_modules/monaco-editor/esm/vs/editor/contrib/semanticTokens/browser/documentSemanticTokens.js
index af9ac92..8fa8632 100644
--- a/node_modules/monaco-editor/esm/vs/editor/contrib/semanticTokens/browser/documentSemanticTokens.js
+++ b/node_modules/monaco-editor/esm/vs/editor/contrib/semanticTokens/browser/documentSemanticTokens.js
@@ -95,8 +95,8 @@ DocumentSemanticTokensFeature = __decorate([
 export { DocumentSemanticTokensFeature };
 let ModelSemanticColoring = class ModelSemanticColoring extends Disposable {
     static { ModelSemanticColoring_1 = this; }
-    static { this.REQUEST_MIN_DELAY = 300; }
-    static { this.REQUEST_MAX_DELAY = 2000; }
+    static { this.REQUEST_MIN_DELAY = 25; }
+    static { this.REQUEST_MAX_DELAY = 100; }
     constructor(model, _semanticTokensStylingService, themeService, languageFeatureDebounceService, languageFeaturesService) {
         super();
         this._semanticTokensStylingService = _semanticTokensStylingService;
@@ -110,9 +110,8 @@ let ModelSemanticColoring = class ModelSemanticColoring extends Disposable {
         this._documentProvidersChangeListeners = [];
         this._providersChangedDuringRequest = false;
         this._register(this._model.onDidChangeContent(() => {
-            if (!this._fetchDocumentSemanticTokens.isScheduled()) {
-                this._fetchDocumentSemanticTokens.schedule(this._debounceInformation.get(this._model));
-            }
+            this._fetchDocumentSemanticTokens.cancel();
+            queueMicrotask(() => this._fetchDocumentSemanticTokens.onTimeout());
         }));
         this._register(this._model.onDidChangeAttached(() => {
             if (!this._fetchDocumentSemanticTokens.isScheduled()) {
@@ -151,14 +150,15 @@ let ModelSemanticColoring = class ModelSemanticColoring extends Disposable {
         bindDocumentChangeListeners();
         this._register(this._provider.onDidChange(() => {
             bindDocumentChangeListeners();
-            this._fetchDocumentSemanticTokens.schedule(this._debounceInformation.get(this._model));
+            this._fetchDocumentSemanticTokens.cancel();
+            queueMicrotask(() => this._fetchDocumentSemanticTokens.onTimeout());
         }));
         this._register(themeService.onDidColorThemeChange(_ => {
             // clear out existing tokens
             this._setDocumentSemanticTokens(null, null, null, []);
             this._fetchDocumentSemanticTokens.schedule(this._debounceInformation.get(this._model));
         }));
-        this._fetchDocumentSemanticTokens.schedule(0);
+        queueMicrotask(() => this._fetchDocumentSemanticTokens.onTimeout());
     }
     dispose() {
         if (this._currentDocumentResponse) {
